### ISSUE TITLE
survey.Required validation not compatible with survey.Confirm prompt

### ISSUE THREAD

**Issue by coolgeek6667** [AlecAivazis/survey#120]
Replicated behavior on both Windows & Linux using the following code:

```
package main

import (
  "fmt"
  "gopkg.in/AlecAivazis/survey.v1"
)

func main() {
  continue := false
  survey.AskOne(
    &survey.Confirm{
      Message: "Continue?",
    },
    &continue,
    survey.Required,
  )
  fmt.Println("continue", continue)
}
```

Expected to see output corresponding to the users input (true/false)

Actual behavior is if the user inputs n/N/No, an error is displayed: *"Sorry, your reply was invalid: Value is required"*.

**Comment by AlecAivazis** [comment]
Thanks for reporting this @coolgeek6667. It's definitely an unexpected consequence of the zero value for a boolean being `false`. I think it would make sense to check if the type we are validating against is a boolean, and having it pass through `survey.Required` without causing an error. While this is slightly off of the documented behavior (since `false` **is** the zero value), I think it makes sense from the mindset that the prompt type might change but you requiring a response does not.

I am currently swamped trying to get out some work for a client as quickly as possible, but this should be a great first issue for someone looking to contribute to `survey`. If you're interested, I can provide some guidance. Also, feel free to reach out to me on the gophers slack channel if you need a more direct line of communication.

**Comment by AlecAivazis** [comment]
@coolgeek6667 - i fixed this in PR #123. Please let me know if that addresses your issue
